ouroboros-network/src/Ouroboros/Network/Protocol/TxSubmission/Codec.hs

```diff
@@ -139,3 +140,20 @@ codecTxSubmission encodeTxId decodeTxId
         (ClientAgency TokTxs,     _, _) ->
           fail "codecTxSubmission.Tx: unexpected message key"
 
+codecTxSubmissionId
+  :: forall txid tx m. Monad m
+  => Codec (TxSubmission txid tx) CodecFailure m (AnyMessage (TxSubmission txid tx))
+codecTxSubmissionId = Codec encode decode
+ where
+  encode :: forall (pr :: PeerRole) st st'.
+            PeerHasAgency pr st
+         -> Message (TxSubmission txid tx) st st'
+         -> AnyMessage (TxSubmission txid tx)
+  encode _ = AnyMessage
+
+  decode :: forall (pr :: PeerRole) st.
+            PeerHasAgency pr st
+         -> m (DecodeStep (AnyMessage (TxSubmission txid tx))
+                          CodecFailure m (SomeMessage st))
+  decode stok = return $ DecodePartial $ \bytes -> case (stok, bytes) of
+    _            -> error "TODO"
```

Comment:
@coot Can someone from the network team implement this? (Not now, next week or so)

Reply:
Did it myself.